### PR TITLE
Enabling XCode build

### PR DESCRIPTION
**Pull request by jessicafalk**
Adding a step to build the XCode project, if we build for iOS

### CODE REVIEW

File: scripts/build-worker.sh [hunk 1]
```diff
@@ -32,13 +32,17 @@ pushd "$(dirname "$0")/../"
...
-        BUILD_TARGET_FILTER_ARG="+buildTargetFilter ${BUILD_TARGET_FILTER}"
+        BUILD_TARGET_FILTER_ARG="+buildTargetFilter ${BUILD_TARGET_FILTER} "
     else
         BLOCK_MESSAGE="Building ${WORKER_TYPE} for ${BUILD_ENVIRONMENT} using ${SCRIPTING_BACKEND}"
         LOG_FILE="$(pwd)/../logs/${WORKER_TYPE}-${BUILD_ENVIRONMENT}-${SCRIPTING_BACKEND}.log"
         BUILD_TARGET_FILTER_ARG=""
     fi
 
+    if [[ -n ${TARGET_IOS_SDK-} ]]; then
+        BUILD_TARGET_FILTER_ARG+="+targetiOSSdk ${TARGET_IOS_SDK}"
```
Comment (jamiebrynes7) [inline scripts/build-worker.sh]: Shouldn't this be its own argument? Rather than tagging onto `BUILD_TARGET_FILTER_ARG` ?

Comment (zeroZshadow) [inline scripts/build-worker.sh]: Should this use a different variable? Instead of reusing the filter

File: scripts/build-worker.sh [hunk 2]
```diff
@@ -55,5 +59,17 @@ pushd "$(dirname "$0")/../"
             +buildEnvironment "${BUILD_ENVIRONMENT}" \
             +scriptingBackend "${SCRIPTING_BACKEND}" \
             "${BUILD_TARGET_FILTER_ARG}"
+
+        if [[ "${BUILD_TARGET_FILTER}" == "ios" ]]; then
```
Comment (jamiebrynes7) [inline scripts/build-worker.sh]: What if `BUILD_TARGET_FILTER` is `ios android` 😉

File: scripts/build-worker.sh [hunk 3]
```diff
@@ -55,5 +59,17 @@ pushd "$(dirname "$0")/../"
...
+            echo "--- Building XCode Project :hammer_and_wrench:"
+
+            dotnet run -p "${RUN_UNITY_PATH}" -- \
+                -projectPath "." \
+                -batchmode \
+                -quit \
+                -logfile "${LOG_FILE}" \
```
Comment (jamiebrynes7) [inline scripts/build-worker.sh]: Should be in its own log file

File: scripts/build-worker.sh [hunk 4]
```diff
@@ -55,5 +59,17 @@ pushd "$(dirname "$0")/../"
...
+                -executeMethod "Improbable.Gdk.Mobile.iOSLaunchUtils.Build" \
```
Comment (jamiebrynes7) [inline scripts/build-worker.sh]: Why is the `Build` method in `iOSLaunchUtils` 🤔

**Review by jamiebrynes7 — CHANGES_REQUESTED** [review]

File: scripts/build-worker.sh (hunk 2 above)
Comment (zeroZshadow) [inline scripts/build-worker.sh]: Does this work if `BUILD_TARGET_FILTER` is not set?

File: scripts/build-worker.sh (hunk 4 above)
Comment (jessicafalk) [reply scripts/build-worker.sh]: cause you first need to build the XCode project, before you can  launch it. 
Happy to change the name if you have a better suggestion

**Review by zeroZshadow — APPROVED** [review]

File: scripts/build-worker.sh (hunk 4 above)
Comment (jamiebrynes7) [reply scripts/build-worker.sh]: `Improbable.Gdk.Mobile.XcodeUtils.Build` and we can throw all the random stuff we need to hack in for XCode in there

File: scripts/build-worker.sh [hunk 5]
```diff
@@ -54,6 +60,19 @@ pushd "$(dirname "$0")/../"
...
             "${BUILD_TARGET_FILTER_ARG}"
+
+        if [[ "${BUILD_TARGET_FILTER-}" =~ "ios" ]]; then
+            echo "--- Building XCode Project :hammer_and_wrench:"
+
+            dotnet run -p "${RUN_UNITY_PATH}" -- \
+                -projectPath "." \
+                -batchmode \
+                -quit \
+                -logfile "$(pwd)/../logs/${WORKER_TYPE}-xcode-build.log" \
```
Comment (jamiebrynes7) [inline scripts/build-worker.sh]:
```suggestion
                -logfile "$(pwd)/../logs/${WORKER_TYPE}-${BUILD_ENVIRONMENT}-xcode-build.log" \
```

Prevent contention between local and cloud :)

File: scripts/build-worker.sh [hunk 6]
```diff
@@ -54,6 +60,19 @@ pushd "$(dirname "$0")/../"
             +buildWorkerTypes "${WORKER_TYPE}" \
             +buildEnvironment "${BUILD_ENVIRONMENT}" \
             +scriptingBackend "${SCRIPTING_BACKEND}" \
+            "${TARGET_IOS_SDK_ARG}" \
             "${BUILD_TARGET_FILTER_ARG}"
+
+        if [[ "${BUILD_TARGET_FILTER-}" =~ "ios" ]]; then
+            echo "--- Building XCode Project :hammer_and_wrench:"
```
Comment (jamiebrynes7) [inline scripts/build-worker.sh]:
```suggestion
            echo "--- Building XCode Project :xcode:"
```

Can't believe you wouldn't check for a custom emoji 😞

**Review by jamiebrynes7 — APPROVED** [review]

File: scripts/build-worker.sh (hunk 6 above)
Comment (jessicafalk) [reply scripts/build-worker.sh]: :O How could I not have done that

File: scripts/build-worker.sh [hunk 7]
```diff
@@ -64,7 +64,7 @@ pushd "$(dirname "$0")/../"
             "${BUILD_TARGET_FILTER_ARG}"
 
         if [[ "${BUILD_TARGET_FILTER-}" =~ "ios" ]]; then
-            echo "--- Building XCode Project :hammer_and_wrench:"
+            echo "--- Building XCode Project :xcode:"
```
Comment (paulbalaji) [inline scripts/build-worker.sh:67]: 💯 was just about to suggest this

**Review by paulbalaji — APPROVED** [review]

**Pull request merged**